components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js

```diff
@@ -1,276 +1,437 @@
+      }
+    },
+    onKeyPress : function(e, regexp, maxLength, padZeros) {
+      var charCode = e.charCode || e.keyCode;
+      var character = String.fromCharCode(charCode);
+      //Call select handler on tab press.
+      if(charCode == 9){
+        this.onValueSelected();
+        return true;
+      }
+      //Allow non-character keys always
+      if (character === "") {return true;}
+      //Block non-digits
+      if (!regexp.test(character)) {return false;}
 
-    populate : function(values) {
-      var selectedIndex = this.dropdown ? (this.dropdown.selectedIndex || this._tmpSelectedIndex) : 0;
+      var currentVal = this.getValue();
+      var caretPos = this.getCaretPosition();
 
-      // using raw HTML for performance reasons: generating many years takes a noticeable time using
-      // more proper methods (e.g. new Element()...)
-      // (Note: using span around select because IE9 does not allow setting innerHTML of <select>-s)
-      var optionsHTML = '<select name="' + this.options.name +
-                             '" class="' + (this.options.cssClass || this.options.name || '') +
-                             '" placeholder="' + (this.options.hint || this.options.name || '') +
-                             '" title="'       + (this.options.hint || this.options.name || '') + '">';
+      currentVal = currentVal.replace(this.getSelectedText(), "");
+      if (currentVal.length == maxLength){return false;}
 
-      optionsHTML += '<option value="" class="empty"></option>';
-      var _this = this;
-      values.each(function (item) {
-         optionsHTML += '<option value="' + item.value + '"';
-         if (item.cssClass) {
-             optionsHTML += ' class="' + item.cssClass + '"';
-         }
-         optionsHTML += '>' + (item.text || item.value || '') + '</option>';
-      });
-      optionsHTML += "</select>";
-      this.span.innerHTML = optionsHTML;
-      this.dropdown = this.span.firstChild;
-      this.callback && this.onSelect(this.callback);
-      if (this.dropdown.selectedIndex <= 0 && selectedIndex >= 0 && selectedIndex < this.dropdown.options.length) {
-        this.dropdown.selectedIndex = selectedIndex;
+      if (padZeros && character != "0"){
+        currentVal = (currentVal || "0") + character; //zero is prepended to character to allow for "2" as valid input
+      } else {
+        currentVal = currentVal.substr(0, caretPos) + character + currentVal.substr(caretPos, currentVal.length);
       }
-    },
 
-    enable : function () {
-      if (!this.options.alwaysEnabled) {
-          this.dropdown.enable();
-          if (this.dropdown.selectedIndex <= 0 && this._tmpSelectedIndex < this.dropdown.options.length) {
-            this.dropdown.selectedIndex = this._tmpSelectedIndex;
-            return (this._tmpSelectedIndex > 0);
-          }
-      }
-      return false;
+      return this.availableValues.map(function(a){return a.indexOf(currentVal) == 0}).reduce(function(a,b){return a || b;});
```

Comment:
I did not mean to say it is not (I checked myself and it was), just mentioned that we need to test all other place sin the code.